spidermon/contrib/actions/sentry/__init__.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from spidermon.contrib.actions.templates import ActionWithTemplates
+from spidermon.exceptions import NotConfigured
+
+
+class SendMessage(ActionWithTemplates):
+    title = None
+    fake = False
+
+    def __init__(self,
+                 title=None,
+                 fake=None,
+                 *args, **kwargs):
+        super(SendMessage, self).__init__(*args, **kwargs)
+        self.title = title or self.title
+        self.fake = fake or self.fake
+
+        if not self.title:
+            raise NotConfigured("You must provide a title for the message.")
+
+    @classmethod
+    def from_crawler_kwargs(cls, crawler):
+        return {
+            'title': crawler.settings.get('SPIDERMON_SENTRY_MESSAGE_TITLE'),
+            'fake': crawler.settings.getbool('SPIDERMON_SENTRY_FAKE'),
+        }
+
+    def run_action(self):
+        message = self.get_message()
+        if not self.fake:
+            self.send_message(message)
+        else:
+            print('-'*40)
+            print(message)
+            print('-'*40)
+
+    def get_title(self):
+        if self.title:
+            return self.render_text_template(self.title)
```

Comment:
This value seems to be always a simple string. Is it really necessary to render it from a template?

Maybe it would be even more clear to remove this method and change L51 to:
`message['title'] = self.title or 'Report Title`

Reply:
We are meant to replace few variables in this string so i would say yes let's do it via the mechanism that we have already.

Reply:
I agree with @muzaffaryousaf on this one, make it extensible it's better. Maybe you want to put the spidername or customer name in the title in a near future.

Reply:
I am still not convinced that the title will be rendered by a template (and inherited from `ActionWithTemplates`), but if we want to follow the pattern available in other similar actions, it would be interesting to include a new setting to allow the user easily provide custom context in the action as:

- https://github.com/scrapinghub/spidermon/blob/master/spidermon/contrib/actions/email/__init__.py#L69
- https://github.com/scrapinghub/spidermon/blob/master/spidermon/contrib/actions/reports/__init__.py#L21

Reply:
@rennerocha Even with available patterns, that's what already happening for different things. e.g. subject. The context that you have shared is only being used for something to include in report itself not for titles / subjects or something for external to the report.